maestro-cli/src/main/java/maestro/cli/device/DeviceCreateUtil.kt

```diff
@@ -0,0 +1,145 @@
+
+        return Device.AvailableForLaunch(
+            modelId = deviceUUID,
+            description = deviceName,
+            platform = Platform.IOS
+        )
+
+    }
+
+    private fun getOrCreateAndroidDevice(version: Int?, forceCreate: Boolean): Device.AvailableForLaunch {
+        if (version !in DeviceConfigAndroid.versions) {
+            throw CliError("Provided Android version is not supported. Please use one of ${DeviceConfigAndroid.versions}")
+        }
+
+        val systemImage = DeviceConfigAndroid.systemImages[version]
+            ?: throw CliError("Provided system image is not supported. Please use one of ${DeviceConfigAndroid.versions}")
+
+        val name = DeviceConfigAndroid.generateDeviceName(version!!)
+
+        // check connected device
+        if (DeviceService.isDeviceConnected(name, Platform.ANDROID) != null) {
+            throw CliError("A device with name $name is already connected")
+        }
+
+        // existing device
+        val existingDevice =
+            if (forceCreate) null else DeviceService.isDeviceAvailableToLaunch(name, Platform.ANDROID)?.modelId
+
+        // dependencies
+        if (existingDevice == null && !DeviceService.isAndroidSystemImageInstalled(systemImage)) {
+            PrintUtils.err("The required system image $systemImage is not installed.")
+
+            PrintUtils.message("Would you like to install it? y/n")
+            val r = readlnOrNull()?.lowercase()
+            if (r == "y" || r == "yes") {
+                PrintUtils.message("Attempting to install $systemImage via Android SDK Manager...\n")
+                if (!DeviceService.installAndroidSystemImage(systemImage)) {
+                    throw CliError("Was unable to install required dependencies.")
+                }
+            } else {
+                throw CliError(
+                    "To install the system image manually, you can run this command:\n${
+                        DeviceService.getAndroidSystemImageInstallCommand(
+                            systemImage
+                        )
+                    }"
+                )
+            }
+        }
+
+        if (existingDevice != null) PrintUtils.message("Using existing device $name.")
+        else PrintUtils.message("Attempting to create Android emulator: $name ")
+
+        val deviceLaunchId = try {
+            existingDevice ?: DeviceService.createAndroidDevice(
+                deviceName = name,
+                device = DeviceConfigAndroid.device,
+                systemImage = systemImage,
+                tag = DeviceConfigAndroid.tag,
+                abi = DeviceConfigAndroid.abi,
+                force = forceCreate
+            )
+        } catch (e: IllegalStateException) {
+            throw CliError("${e.message}")
+        }
+
+        if (existingDevice == null) PrintUtils.message("Created Android emulator: $name ($systemImage)")
```

Comment:
Is `$systemImage` the same as `deviceId` that we reference in maestro CLI and that users can use to specify which device to run on? 

```
      --udid, --device=<deviceId>
                  (Optional) Select a device to run on explicitly
```